api/views.py

```diff
@@ -310,27 +317,30 @@ def post(self, request, *args, **kwargs):
 class GetAuthToken(PublicJsonPostView):
     def handle_post(self, request, *args, **kwargs):
         body = json.loads(request.body.decode('utf-8'))
-        if not 'username' in body or not 'password' in body:
+        if 'username' not in body or 'password' not in body:
             return bad_request('Body must contain `username` and `password`')
         username = body['username']
         password = body['password']
 
-        #allowing different lower/uppercase lettered usernames:
-        try:
-            case_sensitive_username = User.objects.get(username__iexact=username)
-        except User.DoesNotExist:
-            case_sensitive_username = None
-
-        user = authenticate(username=case_sensitive_username, password=password)
+        # Get the case-correct username for authenticate()
+        casecorr_uname = User.objects.filter(username__iexact=username).values_list('username', flat=True).first()
+        if not casecorr_uname:
+            return bad_request('Invalid username or password')  # definately username issue
 
+        # User model's __str__ is its username
+        user = authenticate(username=casecorr_uname, password=password)
         if user is not None:
             api_key, created = Token.objects.get_or_create(user=user)
 
-            # reset the key's created_at time each time we get new credentials
+            # Reset the key's created_at time each time we get new credentials
             if not created:
                 api_key.created = datetime.utcnow().replace(tzinfo=pytz.utc)
                 api_key.save()
 
+            # (Re)set the user's last frontend login datetime
+            user.profile.last_frontend_login = datetime.utcnow().replace(tzinfo=pytz.utc)
```

Comment:
FYI you can use Django timezone to get aware datetime directly.
```
from django.utils import timezone
now = timezone.now()
```
```suggestion
            user.profile.last_frontend_login = timezone.now()
```
https://docs.djangoproject.com/en/3.0/topics/i18n/timezones/#naive-and-aware-datetime-objects

Reply:
Changed it on another place too and removed `pytz` since it's not needed after that. https://github.com/IFRCGo/go-api/pull/808/commits/3cc02486d12465d080533fbc09c7d712ac212933